packages/arianee-access-token/src/lib/arianee-access-token.ts

```diff
@@ -1,24 +1,31 @@
 import { Core } from '@arianee/core';
 import { MemoryStorage } from '@arianee/utils';
 import { ethers, JsonRpcProvider } from 'ethers';
+import memoizee from 'memoizee';
 
 import { JWTGeneric } from './helpers/jwtGeneric';
 import { ArianeeAccessTokenPayload } from './types/ArianeeAccessTokenPayload';
 import { JwtHeaderInterface } from './types/JwtHeaderInterface';
-
 export interface PayloadOverride {
   exp?: number;
   iat?: number;
   [key: string]: string | number | object | undefined;
 }
 
-const ETHEREUM_PROVIDER = 'https://ethereum.arianee.net/';
+const JSON_RPC_PROVIDER = new JsonRpcProvider(
+  'https://eth-mainnet.public.blastapi.io'
+);
 
 export class ArianeeAccessToken {
   private storage: Storage;
 
   private static readonly LAST_AAT_KEY = 'arianee__lastAAT';
 
+  private static memoizedResolveAddress = memoizee(ethers.resolveAddress, {
```

Comment:
ca regle le prob la première fois, mais pas la deuxième fois?

Reply:
first resolving is "slow" (about 1.5s vs 3s prior, we can't really reduce it unless we rewrite the ethers part of it), next resolves for this ENS name are almost instantaneous